examples/seismic/tutorials/12_pickle_dask_op.ipynb

```diff
@@ -0,0 +1,649 @@
+    "    worker to get the current model.\n",
+    "    \"\"\"\n",
+    "    pkl = pickle.load(open(filename, \"rb\"))\n",
+    "    \n",
+    "    return pkl['model']\n",
+    "\n",
+    "def dump_model(filename, model):\n",
+    "    ''' Dump model to disk.\n",
+    "    '''\n",
+    "    pickle.dump({'model':model}, open(filename, \"wb\"))\n",
+    "    \n",
+    "def load_shot_data(shot_id, dt):\n",
+    "    ''' Load shot data from disk, resampling to the model time step.\n",
+    "    '''\n",
+    "    pkl = pickle.load(open(\"shot_%d.p\"%shot_id, \"rb\"))\n",
+    "    \n",
+    "    return pkl['geometry'], pkl['rec'].resample(dt)\n",
+    "\n",
+    "def dump_shot_data(shot_id, rec, geometry):\n",
+    "    ''' Dump shot data to disk.\n",
+    "    '''\n",
+    "    pickle.dump({'rec':rec, 'geometry': geometry}, open('shot_%d.p'%shot_id, \"wb\"))\n",
+    "    \n",
+    "def generate_shotdata_i(param):\n",
+    "    \"\"\" Inversion crime alert! Here the worker is creating the\n",
+    "        'observed' data using the real model. For a real case\n",
+    "        the worker would be reading seismic data from disk.\n",
+    "    \"\"\"\n",
+    "    # Reconstruct objects\n",
+    "    with open(\"arguments.pkl\", \"rb\") as cp_file:\n",
+    "        cp = pickle.load(cp_file)\n",
+    "        \n",
+    "    solver = cp['solver']\n",
+    "\n",
+    "    # source position changes according to the index\n",
+    "    shot_id=param['shot_id']\n",
+    "    \n",
+    "    solver.geometry.src_positions[0,:]=[20, shot_id*1000./(param['nshots']-1)]\n",
+    "    true_d = solver.forward()[0]\n",
+    "    dump_shot_data(shot_id, true_d.resample(4.0), solver.geometry.src_positions)\n",
+    "\n",
+    "def generate_shotdata(solver):\n",
+    "    # Pick devito objects (save on disk)\n",
+    "    cp = {'solver': solver}\n",
+    "    with open(\"arguments.pkl\", \"wb\") as cp_file:\n",
+    "        pickle.dump(cp, cp_file) \n",
+    "\n",
+    "    work = [dict(param) for i in range(param['nshots'])]\n",
+    "    # synthetic data is generated here twice: serial(loop below) and parallel (via dask map functionality) \n",
+    "    for i in  range(param['nshots']):\n",
+    "        work[i]['shot_id'] = i\n",
+    "        generate_shotdata_i(work[i])\n",
+    "\n",
+    "    # Map worklist to cluster, We pass our function and the dictionary to the map() function of the client\n",
+    "    # This returns a list of futures that represents each task\n",
+    "    futures = c.map(generate_shotdata_i, work)\n",
+    "\n",
+    "    # Wait for all futures\n",
+    "    wait(futures)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/muriloboratto/anaconda3/envs/devito/lib/python3.8/site-packages/distributed/node.py:151: UserWarning: Port 8787 is already in use.\n",
```

Comment:
this should be dropped. Could you look for all messages like this and purge the notebook ?

Reply:
done